reports.go

```diff
@@ -46,15 +70,711 @@ type Report struct {
+		if resp.Body != nil {
+			resp.Body.Close()
+		}
+	}()
+
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	getResp := &EducationVerification{}
+	err = json.Unmarshal(b, getResp)
+	if err != nil {
+		return nil, err
+	}
+
+	return getResp, nil
+}
+
+const employmentVerificationSearchPath = "/v1/employment_verifications"
+
+// GetEmploymentVerificationSearch ...
+func (r *Report) GetEmploymentVerificationSearch(employmentVerificationID string, c *Client) (*EmploymentVerification, error) {
+
+	if employmentVerificationID == "" {
+		return nil, errEmptyID
+	}
+
+	rel, err := url.Parse(employmentVerificationSearchPath)
+	if err != nil {
+		return nil, err
+	}
+
+	u := *c.BaseURL
+	u.Path = path.Join(u.Path, rel.String(), employmentVerificationID)
+
+	req, err := http.NewRequest(http.MethodGet, u.String(), nil)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Add("Content-Type", "application/json")
+	req.SetBasicAuth(c.APIKey, "")
+
+	resp, err := c.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, NewResponseError([]int{
+			resp.StatusCode,
+		}, resp)
+	}
+	defer func() {
+		if resp.Body != nil {
+			resp.Body.Close()
+		}
+	}()
+
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	getResp := &EmploymentVerification{}
+	err = json.Unmarshal(b, getResp)
+	if err != nil {
+		return nil, err
+	}
+
+	return getResp, nil
+}
+
+const identityDocumentVerificationSearchPath = "/v1/identity_document_verifications"
+
+// GetIdentityDocumentSearch ...
+func (r *Report) GetIdentityDocumentSearch(identityDocumentVerificationID string, c *Client) (*IdentityDocumentVerification, error) {
+
+	if identityDocumentVerificationID == "" {
+		return nil, errEmptyID
+	}
+
+	rel, err := url.Parse(identityDocumentVerificationSearchPath)
+	if err != nil {
+		return nil, err
+	}
+
+	u := *c.BaseURL
+	u.Path = path.Join(u.Path, rel.String(), identityDocumentVerificationID)
+
+	req, err := http.NewRequest(http.MethodGet, u.String(), nil)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Add("Content-Type", "application/json")
+	req.SetBasicAuth(c.APIKey, "")
+
+	resp, err := c.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, NewResponseError([]int{
+			resp.StatusCode,
+		}, resp)
+	}
+	defer func() {
+		if resp.Body != nil {
+			resp.Body.Close()
+		}
+	}()
+
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	getResp := &IdentityDocumentVerification{}
+	err = json.Unmarshal(b, getResp)
+	if err != nil {
+		return nil, err
+	}
+
+	return getResp, nil
+}
+
+// GetScreenings returns all the screenings for a report
+func (r *Report) GetScreenings(c *Client) (*Screenings, error) {
+	cs := &Screenings{}
+	var err error
+	var wg sync.WaitGroup
+
+	errorChan := make(chan error, 9+len(r.CountyCriminalSearchIDs)+len(r.StateCriminalSearchIDs))
```

Comment:
@chandradeepak I don' think you should buffer the channel since you're closing it anyway.